Market Tools/macd_strategy.py

```diff
@@ -0,0 +1,147 @@
+# Import the required libraries
+import numpy as np
+import pandas as pd
+import hvplot.pandas
+from pathlib import Path
+from datetime import date
+from pandas.core.frame import DataFrame
+import yfinance as yf
+import os
```

Comment:
[Low priority, not for this PR]: Not all of these imports are used in the code.  It's a good practice to remove unused imports. This can make the code easier to read and also prevent errors when running on machines that may not have the unneded package installed.  Lower priority items like this would normally be resolved by adding a backlog item on the Kanban board.